gameplay.py

```diff
@@ -90,8 +106,89 @@ def apply_move_rules(self, attacking_pokemon: Pokemon, move_used: Move, trainer_
+            if ailment == Ailment.UNKNOWN:
+                # Freeze was the randomly chosen ailment and we don't support that, so return
+                return
+        if ailment == Ailment.BURN:
+            # Fire-type Pokemon cannot be burned by a Fire-type move
+            if Type.FIRE not in defending_pokemon.species.types or move_used.info.type != Type.FIRE:
+                print(f"{defending_pokemon.nickname} is burned by the attack!")
+                defending_pokemon.statuses.add(PokemonStatus.BURNED)
+                # TODO: Halve its Attack
+        elif ailment == Ailment.PARALYSIS:
+            # Ground-type Pokemon cannot be paralyzed by an Electric-type move
+            if (
+                Type.GROUND not in defending_pokemon.species.types
+                or move_used.info.type != Type.ELECTRIC
+            ):
+                print(f"{defending_pokemon.nickname} is paralyzed by the attack!")
+                defending_pokemon.statuses.add(PokemonStatus.PARALYZED)
+                # TODO: Decrease Speed by 75%
+        elif ailment == Ailment.POISON:
+            # Poison-type Pokemon cannot be poisoned
+            if Type.POISON not in defending_pokemon.species.types:
+                if move_used.info.api_id == 92:
+                    # The "Toxic" move was used
+                    print(f"{defending_pokemon.nickname} is badly poisoned by the attack!")
+                    defending_pokemon.statuses.add(PokemonStatus.BADLY_POISONED)
+                else:
+                    print(f"{defending_pokemon.nickname} is poisoned by the attack!")
+                    defending_pokemon.statuses.add(PokemonStatus.POISONED)
+        elif ailment == Ailment.CONFUSION:
+            print(f"{defending_pokemon.nickname} is confused by the attack!")
+            defending_pokemon.statuses.add(PokemonStatus.CONFUSED)
+            defending_pokemon.confusion_turns = random.randint(1, 5)
+        # Pokemon can only be bound by one binding move at a time
+        elif ailment == Ailment.TRAP and PokemonStatus.BOUND not in defending_pokemon.statuses:
+            print(f"{defending_pokemon.nickname} is trapped by the attack!")
+            defending_pokemon.statuses.add(PokemonStatus.BOUND)
+            rand_val = random.random()
+            if rand_val < 0.375:
+                defending_pokemon.bound_turns = 2
+            elif rand_val < 0.75:
+                defending_pokemon.bound_turns = 3
+            elif rand_val < 0.875:
+                defending_pokemon.bound_turns = 4
+            else:
+                defending_pokemon.bound_turns = 5
+
     def use_move(self, attacking_pokemon: Pokemon, defending_pokemon: Pokemon,
                  move_to_use: Move, trainer_ind: int):
+
+        if attacking_pokemon.has_status(PokemonStatus.PARALYZED) and random.random() < 0.25:
+            print(f"{attacking_pokemon.nickname} is fully paralyzed! It can't move!")
+            return
```

Comment:
All the returns here happen before the pp is decremented. So I would put the line:
`move_used = move_to_use.use()`

before all of this status logic, so the original used move has its pp decremented

Reply:
Oh shoot, didn't realize pp is still decremented